bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java

```diff
@@ -254,6 +254,16 @@ private String getErrorMessage() {
         return errorMsg;
     }
 
+    private boolean isEmptyDetail() {
+        if (details == null) {
+            return true;
+        }
+        if ((details instanceof MapValue) && ((MapValue) details).isEmpty()) {
```

Comment:
Not changed by this PR but `details` is either a map or a record right? 

@SupunS, @warunalakshitha, can't details be a MapValue instead of Object?

Reply:
Yes, some generic empty check is needed.

Reply:
Yes, it should be a MapValue. This constructor as well as the API in `BValueCreator` are wrong i think.

Reply:
Yeah. 

@KRVPerera, shall we create an issue to track fixing the type of `details` in `ErrorValue`?

We can remove this instanceof check when `details` is defined as a `MapValue` (I guess we can do that now also, but maybe we should keep it just to be on the safer side).

Reply:
Create an issue : #21957